src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -469,4 +515,139 @@ TEST_CASE("request_rate_streaming: test that streaming-specific logic works")
   }
 }
 
+TEST_CASE(
+    "custom_json_data: Check custom json data to ensure that it is processed "
+    "correctly")
+{
+  PerfAnalyzerParameters params{};
+  TestRequestRateManager trrm{params};
+
+  std::shared_ptr<MockModelParser> mmp{
+      std::make_shared<MockModelParser>(false, false)};
+  ModelTensor model_tensor{};
+  model_tensor.datatype_ = "INT32";
+  model_tensor.is_optional_ = false;
+  model_tensor.is_shape_tensor_ = false;
+  model_tensor.name_ = "INPUT0";
+  model_tensor.shape_ = {1};
+  mmp->inputs_ = std::make_shared<ModelTensorMap>();
+  (*mmp->inputs_)[model_tensor.name_] = model_tensor;
+
+  std::shared_ptr<MockDataLoader> mdl{std::make_shared<MockDataLoader>()};
+  const std::string json_str{R"(
+{
+  "data": [
+    {
+      "INPUT0": [2123456789]
+    }
+  ]
+}
+    )"};
+  mdl->ReadDataFromStr(json_str, mmp->Inputs(), mmp->Outputs());
+
+  trrm.parser_ = mmp;
+  trrm.data_loader_ = mdl;
+  trrm.using_json_data_ = true;
+  trrm.execute_ = true;
+  trrm.schedule_.push_back(std::chrono::nanoseconds(1));
+
+  std::shared_ptr<TestRequestRateManager::ThreadStat> thread_stat{
+      std::make_shared<TestRequestRateManager::ThreadStat>()};
+  std::shared_ptr<TestRequestRateManager::ThreadConfig> thread_config{
+      std::make_shared<TestRequestRateManager::ThreadConfig>(0, 0)};
+
+  std::future<void> infer_future{std::async(
+      &TestRequestRateManager::Infer, &trrm, thread_stat, thread_config)};
+
+  early_exit = true;
+  infer_future.get();
+
+  REQUIRE(mdl->recorded_inputs_.size() > 0);
+  CHECK(mdl->recorded_inputs_[0] == 2123456789);
+}
+
+/// Check that the using_shared_memory_ is being set correctly
+///
+TEST_CASE("InitSharedMemory should not be called")
+{
+  PerfAnalyzerParameters params;
+  bool is_sequence = false;
+  bool is_decoupled = false;
+  bool use_mock_infer = true;
+
+  SUBCASE("No shared memory")
+  {
+    params.shared_memory_type = NO_SHARED_MEMORY;
+    TestRequestRateManager trrm(
+        params, is_sequence, is_decoupled, use_mock_infer);
+    trrm.CheckUsingSharedMemoryValue(false);
+  }
+
+  SUBCASE("System shared memory")
+  {
+    params.shared_memory_type = SYSTEM_SHARED_MEMORY;
+    TestRequestRateManager trrm(
+        params, is_sequence, is_decoupled, use_mock_infer);
+    trrm.CheckUsingSharedMemoryValue(true);
+  }
+}
+
+/// Verify Shared Memory api calls
+///
+TEST_CASE("Shared memory methods")
+{
+  PerfAnalyzerParameters params;
+  bool is_sequence = false;
+  bool is_decoupled = false;
+  bool use_mock_infer = true;
+  uint request_rate = 500;
+  uint duration_ms = 1000;
+
+  std::shared_ptr<MockModelParser> mmp{
```

Comment:
we will probably want a common helper function to create a mock data loader and/or mock model parser (partly to hide all this, and partly for sharing)